### PR TITLE
Added UploadDependencies MOJO

### PR DESCRIPTION
**Pull request by jacksondelametter**

### CODE REVIEW

File: pom.xml
Comment (epwilkins) [inline pom.xml]: A: change tabs -> spaces so spacing is consistent

File: pom.xml
Comment (epwilkins) [inline pom.xml]: Q: Why use the Kokuwa parent instead of Tech Brewery parent?

File: pom.xml
Comment (epwilkins) [inline pom.xml]: C: Should be `org.technologybrewery`. Actually most of the other projects also have some "subgroup" that matches the project name (e.g. `org.technologybrewery.habushu`).  Would be wordy here so not sure what's best.

File: LICENSE
Comment (epwilkins) [inline LICENSE:1]: S: I think we need the License file. Did you mean to replace it instead of delete it?

File: src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java
Comment (epwilkins) [inline src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java]: C: Should be `org.technologybrewery`

File: src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java
Comment (epwilkins) [inline src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java]: A: If we use spaces in the pom, we should use spaces here as well.

File: src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java
Comment (epwilkins) [inline src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java]: C: This doesn't support modules which house multiple Helm charts.  Why not use `getChartDirectories`?

File: src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java
Comment (epwilkins) [inline src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java]: C: Again, we need to account for modules which contain multiple charts. Shouldn't be too much harder. Just `getChartDirectories` then for each set the output directory and call `super.execute`

File: pom.xml
Comment (epwilkins) [inline pom.xml:44]: Q/S: Do you not need to use the tech brewery version for now, since it has the extensions updates?

File: src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java
Comment (epwilkins) [inline src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java]: S:  The charts directory may not exist or may be out-of-date if the `DependencyUpdateMojo` hasn't executed.  We should probably document it, print a log message here pointing the user to the solution, or both.

File: README.md
Comment (epwilkins) [inline README.md]: A:
```suggestion
Example: `mvn com.boozallen.aissemble:helm-dependency-maven-plugin:[version]:upload-dependencies -Did=[repositoryID] -Dtype=[NEXUS | ARTIFACTORY | CHARTMUSEUM] -Durl=[url]`
```

File: README.md
Comment (epwilkins) [inline README.md]: A: This makes it sound like this is a required step to use the plugin. I'd change this title to simply "Building from Source" or something similar.  (And maybe move it to the bottom of the page as it's not applicable in most cases.)

File: pom.xml
Comment (jacksondelametter) [reply pom.xml]: There are some dependencies that do not have versions associated with them. By using the parent, it ensures the versions are the same between the parent helm plugin and this one.

File: src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java
Comment (jacksondelametter) [reply src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java]: Added functionality to incorporate subcharts. Though the isHelmChart method is still needed to not re-deploy previously deployed chart dependencies.

File: src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java
Comment (jacksondelametter) [reply src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java]: This has been added.

File: pom.xml
Comment (jacksondelametter) [reply pom.xml:44]: The forked tech brewery version actually does not change the group ID.

File: pom.xml
Comment (jacksondelametter) [reply pom.xml]: Fixed.

File: pom.xml
Comment (jacksondelametter) [reply pom.xml]: Changed to org.technologybrewery.helmdependency

File: LICENSE
Comment (jacksondelametter) [reply LICENSE:1]: Didnt mean to, added back in.

File: src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java
Comment (jacksondelametter) [reply src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java]: Fixed.

File: src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java
Comment (jacksondelametter) [reply src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java]: Fixed.

File: src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java
Comment (jacksondelametter) [reply src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java]: Added.

File: README.md
Comment (jacksondelametter) [reply README.md]: Changed.

File: README.md
Comment (jacksondelametter) [reply README.md]: Changed.

File: pom.xml
Comment (epwilkins) [reply pom.xml:44]: Ah, right.  If we want to get this released without waiting on the upstream PR approval, we'll want to change the group and do a release under the tech brewery group then change this to tech brewery.  All depends on what our needs are. (We've done this pattern of releasing the changes early under tech brewery a few times now so there are examples of the process if we need to do it.)

File: pom.xml
Comment (epwilkins) [inline pom.xml]: A: I would expect this to either match the helm-maven-plugin version or just be 1.0.0-SNAPSHOT

**Review by epwilkins — APPROVED** [review]

File: pom.xml
Comment (jacksondelametter) [reply pom.xml]: Fixed.

**Pull request merged**
